cmake/Googletest.cmake

```diff
@@ -55,12 +55,13 @@ macro(fetch_googletest)
     # https://gitlab.kitware.com/cmake/cmake/issues/17799
     # FindGtest is buggy when dealing with Debug build.
     if (CMAKE_BUILD_TYPE MATCHES Debug AND GTEST_FOUND MATCHES FALSE)
-        message("Setting GTest libraries with debug...")
-        set(GTEST_INCLUDE_DIRS ${GTEST_ROOT}/include)
+        # FindGTest.cmake is buggy when looking for only debug config (it expects both).
+        # So when in debug we set the required gtest vars to the debug libs it would have
+        # found in the find_package(GTest) above. Then we find again. This will then
+        # properly set all the vars and import targets for just debug.
         set(GTEST_LIBRARY ${GTEST_LIBRARY_DEBUG})
-        set(GTEST_LIBRARIES ${GTEST_LIBRARY})
         set(GTEST_MAIN_LIBRARY ${GTEST_MAIN_LIBRARY_DEBUG})
-        set(GTEST_MAIN_LIBRARIES ${GTEST_MAIN_LIBRARY})
+        find_package(GTest QUIET REQUIRED)
```

Comment:
If you look at FindGTest.cmake around line 199 there is test "if(GTEST_FOUND). On line 53 where we call find_package(GTest) it won't be found. As you can see from our test on line 57 (GTEST_FOUND MATCHES FALSE). So because of that it won't enter that if block (in FindGTest.cmake) where it sets the import library.

Thus we set the other two requires vars (GTEST_INCLUDE_DIR is okay) to the debug ones and then find again. This time it GTEST_FOUND will be true and it will set the import library.

Reply:
Ugh I see. This was driving me nuts............... 

Great find. Thank you.